src/libs/tmdk/src/commands/deploy.ts

```diff
@@ -101,108 +100,102 @@ export const handler = async (argv: Arguments<Options>) => {
     }
   }
 
-  // // TODO revisit: import workspace bucket/role (probably need role for specialized permissions)
+  // TODO revisit: import workspace bucket/role (probably need role for specialized permissions)
   // import component types into workspace
-  if (tmdk_config.hasOwnProperty("component-types")) {
-    console.log("====== Component Types ======");
-    let stillComponentRemaining = true; // FIXME cleaner dependency creation process
-    while (stillComponentRemaining) {
-      stillComponentRemaining = false;
-      for (const componentTypeFile of tmdk_config["component-types"]) {
-        const componentTypeDefinitionStr = syncReadFile(
-          `${dir}/${componentTypeFile}`
-        );
-        const componentTypeDefinition = JSON.parse(componentTypeDefinitionStr);
-        // remove inherited properties
-        const propertyDefinitions = componentTypeDefinition[
-          "propertyDefinitions"
-        ] as object;
-        if (propertyDefinitions != undefined) {
-          const filtered_property_definitions = Object.entries(
-            propertyDefinitions
-          ).reduce((acc, [key, value]) => {
-            if (!value["isInherited"]) {
-              acc[key] = value;
-            }
-            return acc;
-          }, {} as { [key: string]: object });
-          componentTypeDefinition["propertyDefinitions"] =
-            filtered_property_definitions;
-        }
-        // create component type if not exists
-        try {
-          const alreadyExists = await createComponentTypeIfNotExists(
-            workspaceIdStr,
-            componentTypeDefinition
-          );
-          await waitForComponentTypeActive(
-            workspaceIdStr,
-            componentTypeDefinition["componentTypeId"]
-          );
-          if (!alreadyExists) {
-            console.log(
-              `Created component-type: ${componentTypeDefinition["componentTypeId"]}`
-            );
-          }
-        } catch (e) {
-          if (
-            // components are not gaurenteed to be in the correct order; retry
-            e instanceof ValidationException &&
-            e.message.includes("do not exist in workspace")
-          ) {
-            stillComponentRemaining = true;
-          } else {
-            throw new Error(`Failed to create component type. ${e}`);
+  console.log("====== Component Types ======");
+  let stillComponentRemaining = true; // FIXME cleaner dependency creation process
+  while (stillComponentRemaining) {
+    stillComponentRemaining = false;
+    for (const componentTypeFile of tmdk_config["component_types"]) {
+      const componentTypeDefinitionStr = syncReadFile(
+        `${dir}/${componentTypeFile}`
```

Comment:
To ensure tmdk is platform agnostic we should use path.join here
https://www.w3schools.com/nodejs/met_path_join.asp

Reply:
+1, I believe there's multiple instances where this applies

Reply:
There are multiple instances, you can make the changes for just these files or leave a TODO for a future commit